### ISSUE TITLE
Call for test for mobile control

### ISSUE THREAD

**Issue by Eureka22** [Eureka22/agar-clone#20]
Hi guys, I just finished the joystick control for my agar-clone, it should be very intuitive and it should work on most mobile phones and tablets. As I only have limited types of devices, could you tell me how do you feel after playing the game on mobile devices? I appreciate your feedback :)

Here is my screenshot:
![wechat_1435482888](https://cloud.githubusercontent.com/assets/5158896/8395792/7bfc7be4-1db9-11e5-9556-1ed4152f2ed4.jpeg)

**Comment by Eureka22** [comment]
Another thing is that I decided not to use gravity control because in this game we need quite precise control and gravity control cannot meet such requirement. Just let me know what you think :)

**Comment by BasvdW22** [comment]
Just tested on my LG G3 (5,5 inch display) and my iPad 3, it works pretty good! I think there's maybe some lag and a little bit of a delay but I don't know if that's different than the desktop version. One thing I noticed is that the blobs don't show up if they are near the left or right side of my iPad screen when I'm spectating. Also, the canvas is a bit trippy when spectating a big blob on my iPad.

**Comment by Eureka22** [comment]
@BasvdW22 Thank you for your feedback. The delay problem might be because the network is slower than desktop devices. And the graphics problems is because at this moment the rendering is not efficient enough(as you can feel if you play for a little while your mobile phone will be very hot). We will address this issue in our next version's client which uses a different rendering engine. :)

**Comment by BasvdW22** [comment]
@Eureka22 Ah nice! You did a pretty good job on the mobile version! Another thing I should mention: I'm now sitting in a train in the Netherlands and the Internet speed is a bit slow, so after a couple of seconds the client freezes. It also freezes instantly if I type in my nickname, let my keyboard open and press play so my keyboard disappears.

**Comment by Eureka22** [comment]
@BasvdW22 Thanks. The client require pretty high Internet speed ranging from 20KB/s to 80KB/s because the client need to send mouse position 40 times a second and the server need to send the position of all objects (including food) 60 times a second. I will see if we can optimise that. 

I didn't experience the second issue you mentioned. But I know it's annoying to have a keyboard pop up in the game so it may be hard to chat in mobile version.

**Comment by vram4** [comment]
@Eureka22 Actually, the update packet is sent every 50 ms for Ogar, so 20 times a second (it is 40 ms on vanilla though). Not sure about mouse position interval

**Comment by Eureka22** [comment]
@vram4 Oh yes. I forgot there are some interpolation. From my test the download speed is about 20-50KB/s, the upload speed is 10-20 KB/s. Can we further optimise that? 

I am pretty sure some choppy frames on mobile devices are caused by the network problem.

**Comment by Eureka22** [comment]
@BasvdW22 Oh, I forgot to ask you are you visiting my site or your site? My server is in SF, USA so if you are in Netherland and visiting my site, there is no wonder the connection will be slow and you will experience delay.

**Comment by vram4** [comment]
@Eureka22 Yes, ive already made it so that the server sends the data for food once. The client will automatically render it until the data that marks it for removal is sent. We can further optimize the packets by sending name data only once. But im not sure how to do that without affecting server performance. Also i dont think the game should be using 20-50 kb/s, that seems like a lot.

**Comment by Eureka22** [comment]
Maybe I overestimated the bandwidth required. And I did a test Ogar vs vanilla server, both in spectating mode, you can find Ogar is already better than vanilla server. 

![image](https://cloud.githubusercontent.com/assets/5158896/8403782/3542bd50-1e7a-11e5-929b-f8c05acbb9c5.png)

First is vanilla, third is Ogar.

**Comment by m-byte** [comment]
Tried it on my Blackberry Q10 via wifi. It works, however, having the joystick only after tapping on the left side of the screen was a bit confusing to me. Also, the display is a bit weird... 
![img_20150629_101910](https://cloud.githubusercontent.com/assets/753104/8403881/ddfe922a-1e48-11e5-8ed8-0b34d8acf5f5.png)

**Comment by m-byte** [comment]
@vram4 @Eureka22 since mouse interval is set on the client, it should be possible to decrease it a bit when a mobile device is detected.

**Comment by Eureka22** [comment]
@vram4 Thanks for the feedback. The joystick is intentionally designed to disappear if not touching. The display can be tuned to support all resolution. I tuned the parameters according to my tests on iPhone 6. Need more work :)

Wait, your client didn't show the split and feed icon?

**Comment by m-byte** [comment]
Nope. And the joystick only appears when using the left side of the screen. On the righht side, I only get a blue circle that does nothing. As you can probably tell by the image, my resolution is 720x720

**Comment by Eureka22** [comment]
@m-byte Just fixed.

**Comment by BasvdW22** [comment]
@Eureka22 I thought the joystick was intended to be only of the left side, lol. Anyway, that's right @Eureka22 , I was visiting your site from the Netherlands, so that explains the delay. And the freezing with the keyboard also seems network related, at my home network it works just fine. I will pass the site to my friends and ask them for feedback.

**Comment by matrixreal** [comment]
hi
@Eureka22 
thanks for the game
the touch event doesn't seem to work good
i see the joystick but the dot dont follow the joystick
the dot follow only when you touch and release but not when you are touching
also is there a way to disable chat on mobile ?
thanks

**Comment by Eureka22** [comment]
@matrixreal The joystick is on the left half of the screen while split and feed button is on the right. It is intended so. 

You can disable chat on the welcome page.

**Comment by BasvdW22** [comment]
@Eureka22 oh right, so it is intended to be only on the left side, never mind. :sweat_smile:

**Comment by mrmarktyy** [comment]
Hi @Eureka22 , do you know approximately what is the minimum bandwidth requirement for running a agar server and having about 50 players online without feeling laggy? thank you!
